packages/client/src/hooks/use-wallet.ts

```diff
@@ -132,8 +146,15 @@ export default function useWallet(): {
             providerName: 'metamask',
             provider: (window as any).ethereum,
         };
-        mixpanel.track('wallet', walletObj);
+
         setWallet(walletObj);
+
+        try {
+          (<any>walletObj).distinct_id =  account;
```

Comment:
I think we should create a new object here instead of casting `walletObj` to any